src/wallet/asyncrpcoperation_sendmany.cpp

```diff
@@ -748,19 +758,84 @@ bool AsyncRPCOperation_sendmany::main_impl() {
 }
 
 
+void AsyncRPCOperation_sendmany::update_joinsplit_sig(CMutableTransaction& mtx)
+{
+    // Empty output script.
+    CScript scriptCode;
+    CTransaction signTx(mtx);
+    uint256 dataToBeSigned = SignatureHash(scriptCode, signTx, NOT_AN_INPUT, SIGHASH_ALL);
+
+    // Add the signature
+    if (!(crypto_sign_detached(&mtx.joinSplitSig[0], NULL,
+            dataToBeSigned.begin(), 32,
+            joinSplitPrivKey_
+            ) == 0))
+    {
+        throw std::runtime_error("crypto_sign_detached failed");
+    }
+
+    // Sanity check
+    if (!(crypto_sign_verify_detached(&mtx.joinSplitSig[0],
+            dataToBeSigned.begin(), 32,
+            mtx.joinSplitPubKey.begin()
+            ) == 0))
+    {
+        throw std::runtime_error("crypto_sign_verify_detached failed");
+    }
+}
+
 /**
  * Sign and send a raw transaction.
  * Raw transaction as hex string should be in object field "rawtxn"
  */
 void AsyncRPCOperation_sendmany::sign_send_raw_transaction(UniValue obj)
-{   
-    // Sign the raw transaction
+{
     UniValue rawtxnValue = find_value(obj, "rawtxn");
     if (rawtxnValue.isNull()) {
         throw JSONRPCError(RPC_WALLET_ERROR, "Missing hex data for raw transaction");
     }
     std::string rawtxn = rawtxnValue.get_str();
 
+#if ENABLE_PROVING_SERVICE
+    if (!testmode && provingService_.is_configured()) {
+        CDataStream stream(ParseHex(rawtxn), SER_NETWORK, PROTOCOL_VERSION);
+        CTransaction tx;
+        stream >> tx;
+        CMutableTransaction mtx(tx);
+
+        if (mtx.vjoinsplit.size() != witnesses_.size()) {
+            throw std::runtime_error("Invalid usage");
```

Comment:
make error message more descriptive?

Reply:
No need; the line number in any error message will be sufficient, as this error should never reach the user in practice (this is an internal function; the check is just a defensive measure to ensure the developers use it correctly).